src/field/pokemon.ts

```diff
@@ -1350,25 +1348,25 @@ export abstract class Pokemon extends Phaser.GameObjects.Container {
   abstract getBossSegmentIndex(): number;
 
   getMoveset(baseOnly?: boolean): PokemonMove[] {
-    const ret = !baseOnly && this.summonData?.moveset ? this.summonData.moveset : this.moveset;
+    const ret = !baseOnly && this.summonData.moveset.length ? this.summonData.moveset : this.moveset;
 
     // Overrides moveset based on arrays specified in overrides.ts
-    let overrideArray: MoveId | Array<MoveId> = this.isPlayer()
+    let overrideArray: MoveId | MoveId[] = this.isPlayer()
       ? Overrides.MOVESET_OVERRIDE
       : Overrides.ENEMY_MOVESET_OVERRIDE;
 
     overrideArray = coerceArray(overrideArray);
-    if (overrideArray.length > 0) {
-      if (!this.isPlayer()) {
-        this.moveset = [];
-      }
-      overrideArray.forEach((moveId: MoveId, index: number) => {
-        const ppUsed = this.moveset[index]?.ppUsed ?? 0;
-        this.moveset[index] = new PokemonMove(moveId, Math.min(ppUsed, allMoves.get(moveId).pp));
-      });
+    if (overrideArray.length === 0) {
+      return ret;
     }
-
-    return ret;
+    if (!this.isPlayer()) {
```

Comment:
NIT: Use `isEnemy()` instead? Or is that not the same?

```suggestion
    if (this.isEnemy()) {
```